src/culture/rebels.cpp

```diff
@@ -1270,11 +1269,11 @@ bool allow_in_area(sys::state& state, dcon::province_id p, dcon::province_id l,
 	case culture::rebel_area::religion:
 		return prov.get_dominant_religion() == rf.get_religion();
 	case culture::rebel_area::nation_religion:
-		return prov.get_dominant_religion() == rf.get_defection_target().get_religion() && prov.get_province_ownership().get_nation() == location.get_province_ownership().get_nation();
+		return prov.get_dominant_religion() == rf.get_defection_target().get_religion() && prov.get_province_ownership().get_nation() == rf.get_rebellion_within().get_ruler();
 	case culture::rebel_area::nation_culture:
-		return prov.get_dominant_culture() == rf.get_defection_target().get_primary_culture() && prov.get_province_ownership().get_nation() == location.get_province_ownership().get_nation();
+		return prov.get_dominant_culture() == rf.get_defection_target().get_primary_culture() && prov.get_province_ownership().get_nation() == rf.get_rebellion_within().get_ruler();
 	case culture::rebel_area::nation:
-		return prov.get_province_ownership().get_nation() == location.get_province_ownership().get_nation();
```

Comment:
shouldn't this have prevented rebels from leaving the nation already?

Reply:
maybe they were crossing borders because they were area all?

Reply:
they do not leabve the nation, defections cause them to

Reply:
Then you should make sure that, when a defection happens, the rebels in the province go away